test/unit/gateways/stripe_test.rb

```diff
@@ -439,6 +439,17 @@ def test_client_data_submitted_with_purchase_without_email_or_order
     end.respond_with(successful_purchase_response)
   end
 
+  def test_client_data_submitted_with_metadata_in_options
+    stub_comms(@gateway, :ssl_request) do
+      updated_options = @options.merge({:metadata => {:option1 => 'optional', :option2 => 'not required'}, :order_id => "42", :email => "foo@wonderfullyfakedomain.com"})
+      @gateway.purchase(@amount,@credit_card,updated_options)
+    end.check_request do |method, endpoint, data, headers|
+      assert_match(/metadata\[option1\]=optional/, data)
+      assert_match(/metadata\[option2\]=not\+required/, data)
+      assert_match(/metadata\[email\]=foo\%40wonderfullyfakedomain\.com/, data)
+      assert_match(/metadata\[order_id\]=42/, data)
+    end.respond_with(successful_purchase_response)
```

Comment:
Huh, ok. I think I've only ever used it when I've needed the response, so just assumed you didn't need it here.